mantidproject/mantid
# More Flexible LiveListeners

This PR introduces many of the changes planned in the [More Flexible LiveListeners](https://github.com/mantidproject/documents/blob/master/Design/MoreFlexibleILiveListener.md) design document.

Overview of changes:

* New LiveListenerInfo class has been added to Mantid::Kernel
* FacilityInfo and InstrumentInfo classes have been updated to parse new xml syntax
* Facilities.xml has been updated to use new syntax 
* Entries like ENGIN-X and ENGIN-X_EVENT have **NOT** been merged into one instrument yet
* LiveListenerFactory now has two `create` function overloads:
  * Previous version that takes InstrumentName (and now optionally a LiveListenerInfo name)
  * New version that takes a LiveListenerInfo directly (can be used to construct listeners without facilities.xml)
* LiveDataAlgorithm has three new optional properties to override facilities.xml settings: 
  * Connection - Name of `<connection ... />` entry to use
  * Listener - Name of specific LiveListener class to use
  * Address - Address string to use
* StartLiveData dialog has been updated with new controls for above properties
* StartLiveData dialog supports overriding Listener class name and Address string from facilities.xml settings
* Removed connectivity check from MWRunFiles' LiveData Button (preliminary; this control will need rework to support multiple LiveListenerInfo's per InstrumentInfo)

**To test:**

* Ensure StartLiveData dialog changes make sense
* Ensure StartLiveData dialog can still be used for current workflows without issues
* Ensure StartLiveData can still be called from Python without issues

Fixes #18007.



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
